### DISCUSSION TITLE
Why reactable is not working with print() in Quarto

### DISCUSSION THREAD

**Discussion by shafayetShafee** [quarto-dev/quarto-cli#2695] [Q&A] [answered]
While `reactable(iris)` is working as intended, `print(reactable(iris))` is not showing any table in the output. Would anyone kindly help me to understand this behavior or point me to a direction where I would find a detailed discussion about this?

**Reprex**
~~~
---
title: "Printing Reactable"
format: html
---

```{r}
library(reactable)

reactable(iris)
```

```{r}
print(reactable(iris))
```
~~~

Though there's no specific reason to use `print(reactable(iris))` instead of `reactable(iris)`, but while working with `reactable` in Quarto, noticed this behavior and failed to understand why this is happening by myself. Any help is truly appreciated 😊.

**Comment by cderv** [comment, marked as answer]
This is not specific to Quarto, it will be exactly the same with **rmarkdown**. This is due to the way **knitr** is working - **knitr** being the R package used for handling R code chunks within a Quarto document using R. 

`knitr::knit_print()` will be the function used for printing, and method used depends on the object class. For **reactable**, I believe **htmlwidgets** methods will be used. `print()` will not return a printable object for **knitr** - only the `knit_print()` method will. 

The general rule is that `print()` should rarely be called directly in a code chunk as usually we won't know how to handle the printing for inside a knitting. 

More on this in https://bookdown.org/yihui/rmarkdown-cookbook/opts-render.html and https://cran.r-project.org/web/packages/knitr/vignettes/knit_print.html

Hope it helps understand

**Reply by shafayetShafee** [reply]
Oh, got it. Thanks.
